packages/rum/src/boot/postStartStrategy.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  SessionReplayState,
+  type LifeCycle,
+  type RumConfiguration,
+  type RumSessionManager,
+  type StartRecordingOptions,
+  type ViewHistory,
+} from '@datadog/browser-rum-core'
+import { runOnReadyState, type DeflateEncoder } from '@datadog/browser-core'
+import { getSessionReplayLink } from '../domain/getSessionReplayLink'
+import type { startRecording } from './startRecording'
+
+export type StartRecording = typeof startRecording
+
+export const enum RecorderStatus {
+  // The recorder is stopped.
+  Stopped,
+  // The user started the recording while it wasn't possible yet. The recorder should start as soon
+  // as possible.
+  IntentToStart,
+  // The recorder is starting. It does not record anything yet.
+  Starting,
+  // The recorder is started, it records the session.
+  Started,
+}
+export type RecorderState = {
+  status: RecorderStatus
+  setStatus: (status: RecorderStatus) => void
+  statusEqual: (status: RecorderStatus) => boolean
```

Comment:
Nit: since a `RecorderState` must provide a gettable `status` anyway, would it work just as well to remove `statusEqual` and use `status ===` instead?